app/src/main/java/com/example/photofy/DetectProperties.java

```diff
@@ -38,34 +29,18 @@ public class DetectProperties {
 
     public static final String TAG = "DetectProperties";
 
-    private Context context;
     private Photo picture;
     private String path;
     private GoogleCredentials credentials;
     private List<String> objects;
 
-    public DetectProperties(Photo picture, String path, Context context) {
+    public DetectProperties(Photo picture, String path) {
```

Comment:
constructor should not receive the `picture` param. may be rename this class to `ImageDominantColorFinder` as it is only finding dominant color and no other property. 

```
class ImageDominantColorFinder {
  public ImageDominantColorFinder(String credentialsFilePath) {
   // ... create credential object and annotator client.
  //  ImageAnnotatorClient client = ImageAnnotatorClient.create(imageAnnotatorSettings)
  }
  
  String findDominantColor(Photo photo, String path) {
   // upload the image to gcs cloud
  // call annotatorclient to find the color
  return colorSring;
  }
}
```